### PR TITLE
Add JSON output option for buildinfo.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This adds a variant of the `-W buildinfo` option called `-W buildinfojson` which outputs JSON instead of plain text, allowing for easier parsing and simpler handling in our anonymous statistics reporting.

Sample output (with JSON syntax hilighting):

```json
{
  "features": {
    "dbengine": "NO",
    "native-https": "YES",
    "cloud": "NO",
    "tls-host-verify": "YES"
  },
  "libs": {
    "jemalloc": "NO",
    "jsonc": "YES",
    "libcap": "YES",
    "libcrypto": "YES",
    "libm": "YES",
    "lws": "NO",
    "mosquitto": "NO",
    "tcmalloc": "NO",
    "zlib": "YES"
  },
  "plugins": {
    "apps": "YES",
    "cgroup-net": "YES",
    "cups": "YES",
    "ebpf": "NO",
    "ipmi": "NO",
    "nfacct": "YES",
    "perf": "YES",
    "slabinfo": "YES",
    "xen": "NO",
    "xen-vbd-error": "NO"
  },
  "exporters": {
    "kinesis": "NO",
    "pubsub": "NO",
    "mongodb": "NO",
    "prom-remote-write": "YES"
  }
}
```

##### Component Name

area/daemon

##### Test Plan

Manually tested by building with various options and verifying that the output from `netdata -W buildinfo` and `netdata -W buildinfojson` match correctly.

Also validated all the output to ensure it’s valid JSON.

##### Additional Information

The resultant output intentionally does not include the configure options. These are very likely to leak identifying information when dealing with custom builds, and would also require comprehensive escaping to be safe for use as part of  a JSON string.

This also intentionally does not use JSON-C. We need this to work even if we don’t have JSON-C, so a trivial static substitution approach like this is really the only option unless we want to include a different JSON implementation (and I strongly suspect we do not, as much as I would love to just vendor [cJSON](https://github.com/DaveGamble/cJSON) instead of depending on JSON-C as an external dependency).

### CODE REVIEW

**Comment by underhood** [comment]
wouldn't it be better to have bools instead of strings "YES"/"NO"?

**Comment by Ferroin** [comment]
> wouldn't it be better to have JSON bools (`"xyz": true`)instead of strings "YES"/"NO"?

Possibly, but that makes the code somewhat more complicated, and for at least the LWS thing we’d need much more complex handling on top of the extra complexity involved in just making the things boolean values.

**Review by andrewm4894 — DISMISSED** [review]
The output of this looks good to me. I wonder if "true" / "false" might be better then "YES" / "NO" as it would be easier then to just convert them to booleans as needed. But at the same time maybe making a string called "true" has its own issues. 

No strong opinion from me anyway - either approach will work for what we need.

**Comment by MrZammler** [comment]
Hi! I'm ok with this too (though I agree, it's better to use true/false).

However, unless I'm not up-to-date with something, if we want to include the buildinfo details on the statistics, we could avoid calling the binary with a `popen` to get the data, and instead get them from "inside" the agent code (and your code showed me how to do it! :-) ) and provide it to the anonymous-statistics script.

**Comment by underhood** [comment]
@Ferroin take a look at my latest commit if disagree skip it and I approve (sorry for the direct push that was unintended)

File: daemon/buildinfo.c
Comment (underhood) [inline daemon/buildinfo.c:265]: forgot to remove quotes here

File: daemon/buildinfo.c
Comment (underhood) [inline daemon/buildinfo.c:267]: here I forgot to add quotes

**Review by underhood — CHANGES_REQUESTED** [review]

**Comment by thiagoftsm** [comment]
I made some tests now and everything looks ok, but I will wait for @Ferroin to take decision about @underhood suggestions before to approve the PR.

**Comment by Ferroin** [comment]
> @Ferroin take a look at my latest commit if disagree skip it and I approve (sorry for the direct push - that was unintended, planned to push to my fork... )

And this is part of why I’m not a C developer...

The changes you’re suggesting look good to me, I just never would have thought of such an approach myself.

**Comment by underhood** [comment]
@Ferroin just trying to pay back your help with the installer and CI etc. :) I will fix my remaining 2 mistakes and we can go ahead with it. Can I push the fix directly to the branch?

**Comment by Ferroin** [comment]
@underhood Yeah, just push the fixes directly.

**Review by thiagoftsm — APPROVED** [review]

**Review by MrZammler — APPROVED** [review]

**Pull request merged**
